src/net/metaplaylist/isobmff_patcher.ts

```diff
@@ -0,0 +1,461 @@
+            output = this.mvhd(data);
+            break;
+          case "tkhd":
+            output = this.tkhd(data);
+            break;
+          default:
+            output = data;
+            break;
+        }
+        // if (output.toString() !== data.toString()) {
+        //   console.log(boxtype);
+        //   console.log("");
+        // }
+      }
+      return output;
+    }
+
+    tkhd(input: Uint8Array): Uint8Array {
+      const version = input[8];
+      let output = new Uint8Array(0);
+      if (version === 1) {
+        output = concat(
+          input.subarray(0,36),
+          itobe8(0),
+          input.subarray(44, input.length)
+        );
+      } else {
+        output = concat(
+          input.subarray(0,28),
+          itobe4(0),
+          input.subarray(32, input.length)
+        );
+      }
+      return output;
+    }
+
+    mvhd(input: Uint8Array): Uint8Array {
+      const version = input[8];
+      let output = new Uint8Array(0);
+      if (version === 1) {
+        output = concat(
+          input.subarray(0,32),
+          itobe8(0),
+          input.subarray(40, input.length)
+        );
+      } else {
+        output = concat(
+          input.subarray(0,24),
+          itobe4(0),
+          input.subarray(28, input.length)
+        );
+      }
+      return output;
+    }
+
+    /**
+     * Process styp and make sure lmsg presence follows the lmsg flag parameter.
+     * Add scte35 box if appropriate
+     * @param {Uint8Array} input
+     * @return {Uint8Array} patched box
+     */
+    styp(input: Uint8Array): Uint8Array {
+      const lmsg = this.lmsg;
+      const size = be4toi(input, 0);
+      let pos = 8;
+      const brands = [];
+      while (pos < size) {
+          const brand = input.subarray(pos, pos + 4);
+          if (bytesToStr(brand) !== "lmsg") {
+              brands.push(brand);
+          }
+          pos += 4;
+      }
+      if (lmsg) {
+        brands.push(strToBytes("lmsg"));
+      }
+      const dataSize = brands.length * 4;
+      const data = new Uint8Array(dataSize);
+      for (let i = 0; i < brands.length; i++) {
+        data.set(brands[i], i*4);
+      }
+      return Atom("styp", data);
+    }
+
+    /**
+     * Process tfhd (assuming that we know the ttml size size).
+     * @param {Uint8Array} input
+     * @return {Uint8Array} patched box
+     */
+    tfhd(input: Uint8Array): Uint8Array {
+      let output = new Uint8Array(0);
+      if (!this.isTtml) {
+        return input;
+      }
+      const tfFlags = be4toi(input.subarray(8,12), 0) & 0xFFFFFF;
+      let pos = 16;
+      if (tfFlags & 0x01) {
+        throw new Error("base-data-offset-present not supported in ttml segments");
+      }
+      if (tfFlags & 0x02) {
+        pos += 4;
+      }
+      if ((tfFlags & 0x08) === 0) {
+        throw new Error("Cannot handle ttml segments with no default_sample_duration");
+      } else {
+        pos += 4;
+      }
+      if (tfFlags && 0x10) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: This condition always evaluates to "true". ([typescript:S2589](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS2589))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=178&issues=AWNyk0oJTmDVX2bfxyas&open=AWNyk0oJTmDVX2bfxyas)

[ISSUE_KEY]: AWNyk0oJTmDVX2bfxyas